src/workerd/io/observer.h

```diff
@@ -30,6 +30,18 @@ class WebSocketObserver: public kj::Refcounted {
   virtual void receivedMessage(size_t bytes) {};
 };
 
+// Observes a byte stream. This interface should only be used by byte streams which internally queue
+// each write and each write of size N is accompanied by a corresponding read of the same size.
```

Comment:
This second sentence is a bit unclear as to what it means.

Reply:
Have tried to rephrase.